cv32e40s/tests/programs/custom/illegal_access_loop_tests/illegal_custom_loop_gen.py

```diff
@@ -0,0 +1,65 @@
+"""
```

Comment:
The file generated by this script is `~ 130 000` lines long.
That is so big that github refuses to show its content.
I am a bit skeptical to merge such a big file.

Reply:
This is because there are 130K possible custom instructions at the moment. The size of the file is 2.3MB. I don't know how one would deal with this otherwise, I could remove the generated lines but then anyone running the test would have to manually generate them again.

Reply:
I'll also add that the test-time lies between 22 and 36 minutes.

Reply:
I shall read the generation script and see if I can think of any other way to do this.

One thought that I have already:
For the other generation script (the one created that 12k program), self-modifying code _might_ be a solution.
(Though, warning, self-modifying code is probably a bit tricky.)
We know that a CSR instruction uses bit `[31:20]` for the CSR address...
So if you have e.g. 4 different instructions (read/write/set/clear), then we can add labels to them and let the surrounding code overwrite those bits.
Instead of several thousands of lines of code, we'd have a loop that does the same thing.

Reply:
@silabs-hfegran and @MikeOpenHWGroup, this big file is difficult.
Do you have any opinions, insights, or solutions?

Reply:
> I shall read the generation script and see if I can think of any other way to do this.
> 
> One thought that I have already: For the other generation script (the one created that 12k program), self-modifying code _might_ be a solution. (Though, warning, self-modifying code is probably a bit tricky.) We know that a CSR instruction uses bit `[31:20]` for the CSR address... So if you have e.g. 4 different instructions (read/write/set/clear), then we can add labels to them and let the surrounding code overwrite those bits. Instead of several thousands of lines of code, we'd have a loop that does the same thing.

Maybe, but I believe this was discussed and (I might be mistaken here) the conclusion was that the commands need to be loaded into memory at compile time, and you can't (I believe) change the asm command while running. Although, @silabs-hfegran has a better understanding of it.

Reply:
Pseudo assembly to demonstrate some of the idea:
```
my_csr_instr:
  csrrs t0, 0x0, x0
  
  la t1, my_csr_instr
  lw t2, 0(t1)
  t2 &= 0x000F_FFFF
  t2 |= 0x3A00_0000
  sw t2 0(t1)
  j my_csr_instr
```

But as we discussed, we also have the option of using some random instructions instead of every hypothetical one. (Maybe combine these two ideas.)

Reply:
We agreed that the file size is not too big, and we are keeping the current method. Resolved.